Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift

```diff
@@ -0,0 +1,293 @@
+            configuration.worldAlignment = .gravityAndHeading
+        }
+        
+        _locationDataSource = .init(initialValue: locationDataSource)
+    }
+    
+    public var body: some View {
+        SceneViewReader { sceneViewProxy in
+            ZStack {
+                ARSwiftUIView(proxy: arViewProxy)
+                    .onDidUpdateFrame { _, frame in
+                        guard let interfaceOrientation, initialCameraIsSet else { return }
+                        
+                        sceneViewProxy.updateCamera(
+                            frame: frame,
+                            cameraController: cameraController,
+                            orientation: interfaceOrientation,
+                            initialTransformation: .identity
+                        )
+                        sceneViewProxy.setFieldOfView(
+                            for: frame,
+                            orientation: interfaceOrientation
+                        )
+                    }
+                
+                if initialCameraIsSet {
+                    sceneViewBuilder(sceneViewProxy)
+                        .cameraController(cameraController)
+                        .attributionBarHidden(true)
+                        .spaceEffect(.transparent)
+                        .atmosphereEffect(.off)
+                        .interactiveNavigationDisabled(true)
+                        .onCameraChanged { camera in
+                            currentCamera = camera
+                        }
+                }
+                
+                if !coachingOverlayIsHidden {
+                    ARCoachingOverlay(goal: .geoTracking)
+                        .sessionProvider(arViewProxy)
+                        .active(coachingOverlayIsActive)
+                        .allowsHitTesting(false)
+                }
+            }
+        }
+        .observingInterfaceOrientation($interfaceOrientation)
+        .onAppear {
+            arViewProxy.session.run(configuration)
+        }
+        .onDisappear {
+            arViewProxy.session.pause()
+            Task { await locationDataSource.stop() }
+        }
+        .task {
+            do {
+                try await locationDataSource.start()
+                await withTaskGroup(of: Void.self) { group in
+                    group.addTask {
+                        for await location in locationDataSource.locations {
+                            lastLocationTimestamp = location.timestamp
+                            currentLocation = location
+                            await updateSceneView(for: location)
+                        }
+                    }
+                    group.addTask {
+                        for await heading in locationDataSource.headings {
+                            currentHeading = heading
+                            updateHeading(heading)
+                        }
+                    }
```

Comment:
I noticed that with continuous heading updates, the scene view doesn't yaw smoothly when the device is slowly turning. We'll revisit this problem in the following PR where the continuous/initial modes are added.

Reply:
This issue should be fixed now by animating the heading changes.